Add nessus file import and manual finding adding, asset criticality, …

JasperJuergensen opened the pull request:
…tagging

In the finding import form now the engine type can be selected. An engine type 'nessus' is added. The nessus report file then will be parsed and the findings are added via the _import_findings interface.
The nessus has a critical severity in addition to the severities info, low, medium and high. Also most scanning engines uses five severity levels, so a severity level critical is added. In every view, where finding severities are used, this new severity level is added. The color scheme for the severities is updated, so that criticial is now red, high orange, medium yellow, low green and info blue.
#21 

A form for adding a finding manually is added.
#22 

If there are only three criticality levels, it is most likely, that normally the criticality in the middle is choosen. Therefor, a fourth criticality level ('critical') is added.
#23 

The tagging system has been improved. When adding a finding manually, it is possible to add tags in the tag field by pressing the tab-key after entering the tag name. There is also now a edit finding interface, in which it is also possible to add tags.

MaKyOtOx commented:
Hi Jasper,
Thank you very much for your PR !! Impressive work. We have to review and test all the features before integrate to the 'master' branch.

We have several topics to discuss with you:
- Regarding the issue #23, it is not so obvious :/ I'm definitely agree with you to avoid the middle level (no choice). But we have to find a way to harmonise the finding levels collected from all the engines...
In our first studies, only Nessus had 4 levels of findings. 'Critical' and 'high' findings from Nessus are translated to 'high' by the PatrowlEngine. The difference between the 2 levels is very thin and practically useless --> you have to patch either high or critical vulnerabilities with the same urgency.
- However, regarding the green color for low findings, I'm not in your side ;) Green is the common color for good things. A low finding remain a bad thing. Not so impacting, but not right.
We have to assess these considerations before updating the number of levels.
- By the way, if we had another asset criticity level, we have to change all summaries, bars, dashboard graphs and score/grade evaluation. 

Regardless these area of improvement, you provided a great work !! Thanks again

MaKyOtOx commented:
Do you agree if, in a first time, we push the manual import of findings and the "fast-adding" tag features ?

JasperJuergensen commented:
Hi Nicolas,
I see your point for the finding severity levels and the asset criticity levels. I did this according to BSI TR 200-3, which defines four levels for asset criticity, but I'm fine with three levels. 
For findings, I think there is a difference between high and critical findings. CVSS also uses 4 ratings in scoring. 
Green for low I'm all yours, maby this can be colored using the colors of the CVSS online calculator (critical: cc0500, high: df3d03, medium: f9a009 and low: ffcb0d)

I'm currently fixing some bugs, which I found a little late in my nessus importer, so there will be at least one commit more.

Many Greetings
Jasper

JasperJuergensen commented:
I've updated the color scheme. You may have a look at that. 
The asset criticity 'critical' is removed but the finding severity 'critical' is still there. But I've updated the asset grade calculation.

MaKyOtOx commented:
Thank you for the updates ! We will review the changes asap :)

MaKyOtOx commented on findings/forms.py:
```diff
@@ -1,9 +1,45 @@
+			'confidence': forms.TextInput(attrs={'class': 'form-control form-control-sm'}),
+			'severity': forms.Select(attrs={'class': 'form-control form-control-sm'}),
+			'scan': forms.Select(attrs={'class': 'form-control form-control-sm'}),
+			'owner': forms.Select(attrs={'class': 'form-control form-control-sm'}),
```
dangerous :)

MaKyOtOx commented:
Some changes have been suggested. See https://github.com/Patrowl/PatrowlManager/tree/vace-sec-feature/nessus_import

MaKyOtOx commented:
Future improvements have been identified (not required to be merge the PR into master):
- Add/Edit finding view: enhance UX/UI to reduce format errors 
- Asset creation workflow when findings are imported
